contracts/Comptroller.sol

```diff
@@ -365,6 +368,14 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
             return uint(Error.PRICE_ERROR);
         }
 
+        // Error if borrow brings total borrow above limit
+        (MathError mathErr, uint hypotheticalTotalBorrows) = addUInt(CToken(cToken).totalBorrows(),borrowAmount);
```

Comment:
We'll have to ensure that the `totalBorrows()` call here is safe. As I think it's generally a simple storage read, I believe it should be, but want to double check this extra interaction.